### PR TITLE
Update/5 unified collections exhibitions

### LINKED ISSUE #5: Spike: how do we modify the transcoder so we can use separate processes for Collection and Exhibition videos?
Currently the transcoder is geared towards collection videos only. We want a separate pipeline for Exhibition videos, and flexible pipelines in general.

I think we should define pipelines as sets of functions:

- file-generator - a function that generates file paths for input files to process
- processes - an ordered dictionary of lists of functions, each taking as input the output of the function before it.
- finally - final tasks to do with the output of all the processes once they have all completed.

### PR DESCRIPTION
**Pull request by hvanness**
*Resolves #5 and #14*

This is a preliminary step towards having a more configurable transcoder that supports both exhibitions and collections video transcoding.
Adds a FLEXIBLE_MASTER_NAMING boolean environment variable in order to support exhibitions and collections file processing from the same codebase.
Adds documentation for deploying to OSX (without balena).
Includes some general refactor and cleanup.
Removes local web copies after copying them to S3.

To test:
1. Set FLEXIBLE_MASTER_NAMING=False
2. Check that a file named like B9234567_mp01_placeholder-2.mp4 transcodes, and that a file named like placeholder-2.mp4 doesn't transcode.
3. Set FLEXIBLE_MASTER_NAMING=True
4. Check that videos transcode with any name, like placeholder-2.mp4

### CODE REVIEW

File: dev.tmpl.env [hunk 1]
```diff
@@ -19,9 +19,4 @@ XOS_API_ENDPOINT=http://<your ip>:8000/api
 # XOS_AUTH_TOKEN token's user requires staff privileges for PATCH to /api/assets
 XOS_AUTH_TOKEN=AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
 
-# If not using Docker and SMB, set local directories to use
-# WATCH_FOLDER=/Users/stevejobs/acmi/devmount/watch
-# MASTER_FOLDER=/Users/stevejobs/acmi/devmount/master
-# ACCESS_FOLDER=/Users/stevejobs/acmi/devmount/access
-# WEB_FOLDER=/Users/stevejobs/acmi/devmount/web/
-# OUTPUT_FOLDER=/Users/stevejobs/acmi/devmount/output
+FLEXIBLE_MASTER_NAMING=False
```
Comment (sighmon) [inline dev.tmpl.env:22]: Add trailing slash to template: `XOS_API_ENDPOINT=http://<your ip>:8000/api/`

**Review by sighmon — APPROVED** [review]
Looks good and functions as expected. One small change to the `dev.tmpl.env`, and just a note on the output - seems to repeat itself when it errors with the file naming. But not a blocker to get it released.

```bash
ValueError: simon.mp4 is not named like a collections preservation master file. Consider setting the environment variable FLEXIBLE_MASTER_NAMING=True.
transcoder_1  | 2020-03-18 11:24:49: WARNING - Could not make sure we have the destination folders. There may be something funny with the file name: simon.mp4 is not named like a collections preservation master file. Consider setting the environment variable FLEXIBLE_MASTER_NAMING=True.
```

**Comment by sighmon** [comment]
**General question**: is there a reason why web versions aren't saved to the `/mount/web` folder?

**Comment by hvanness** [comment]
> Looks good and functions as expected. One small change to the `dev.tmpl.env`, and just a note on the output - seems to repeat itself when it errors with the file naming. But not a blocker to get it released.
> 
> ```shell
> ValueError: simon.mp4 is not named like a collections preservation master file. Consider setting the environment variable FLEXIBLE_MASTER_NAMING=True.
> transcoder_1  | 2020-03-18 11:24:49: WARNING - Could not make sure we have the destination folders. There may be something funny with the file name: simon.mp4 is not named like a collections preservation master file. Consider setting the environment variable FLEXIBLE_MASTER_NAMING=True.
> ```

Hm, post_slack_exception prints the traceback with `traceback.print_exc()` which is good, maybe this should have replaced the `logging.warning(message)` line. Making the change.
```
def post_slack_exception(message):
    traceback.print_exc()
    logging.warning(message)
    post_slack_message(message)
```

File: dev.tmpl.env (hunk 1 above)
Comment (hvanness) [reply dev.tmpl.env:22]: Good call, done.

**Comment by hvanness** [comment]
> **General question**: is there a reason why web versions aren't saved to the `/mount/web` folder?

Greg suggested only saving web versions on S3 in order to save in-house disk space.

**Pull request merged**